.github/workflows/weekly.yml

```diff
@@ -0,0 +1,71 @@
+# Weekly tests to catch changes in upstream dependencies like pins
+# that break vetiver e.g. changes in pins could break either one or
+# both of:
+#   1. pypi release
+#   2. main branch version
+
+name: Weekly Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    # At 00:00 on Monday
+    - cron: 0 0 * * MON
+
+jobs:
+  vetiver_main_pins_main:
+    name: 'vetiver main, pins main'
+    if: github.repository == 'tidymodels/vetiver-python'
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # Minimum and Maximum supported versions
+        python-version: ['3.7', '3.10']
```

Comment:
Will the min and max supported version also capture enough info to check in-between versions?